Added feature request for origin_path in aws_cloudfront_distribution … (#516)
inspec/inspec-aws: docs/resources/aws_cloudfront_distribution.md

For additional information, see the [AWS API reference for CloudFront distributions](https://docs.aws.amazon.com/cloudfront/latest/APIReference/API_distribution.html) documentation. For available SSL/TLS version identifiers, see [OriginSslProtocols](https://docs.aws.amazon.com/cloudfront/latest/APIReference/API_OriginSslProtocols.html) and [AWS::CloudFront::distribution ViewerCertificate](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-cloudfront-distribution-viewercertificate.html) documentation.

### origin\_domain\_name _(optional)_

> 
> If provided, this parameter is expected an existing `origin_domain_name` along with `s3_origin_path`
> 
> It will return empty string if `s3_origin_path` path wasn't configured, else it will retuen the configured path.
> 
> 

Reviewer: ```suggestion

The domain name for the origin.

Provide the `origin_domain_name` if you want to validate the `s3_origin_path` property.

```

|custom\_origin\_ssl\_protocols              | An array containing SSL/TLS protocols allowed by custom origins in this distribution. Empty if there are no custom origins (one or more standard S3 bucket origins). Current valid values are `SSLv3`, `TLSv1`, `TLSv1.1`, `TLSv1.2`. |
|viewer\_certificate\_minimum\_ssl\_protocol | The minimum SSL/TLS protocol version in the Viewer Certificate. Current valid values: `SSLv3`, `TLSv1`, `TLSv1_2016`, `TLSv1.1_2016`, `TLSv1.2_2018`, `TLSv1.2_2019`, `TLSv1.2_2021`. |
|s3\_origin\_config                          | `True`: if there are any S3 origin configs in the distribution (i.e. standard S3 bucket origins), else `False`. |

> |origin\_domain\_name                        | Provide the `origin_domain_name` in case want to validate `s3_origin_path`

Reviewer: Is the `origin_domain_name` one of the properties, or is it just a parameter?

Author: it is an optional parameter. for this change property is `s3_origin_path`

|custom\_origin\_ssl\_protocols              | An array containing SSL/TLS protocols allowed by custom origins in this distribution. Empty if there are no custom origins (one or more standard S3 bucket origins). Current valid values are `SSLv3`, `TLSv1`, `TLSv1.1`, `TLSv1.2`. |
|viewer\_certificate\_minimum\_ssl\_protocol | The minimum SSL/TLS protocol version in the Viewer Certificate. Current valid values: `SSLv3`, `TLSv1`, `TLSv1_2016`, `TLSv1.1_2016`, `TLSv1.2_2018`, `TLSv1.2_2019`, `TLSv1.2_2021`. |
|s3\_origin\_config                          | `True`: if there are any S3 origin configs in the distribution (i.e. standard S3 bucket origins), else `False`. |

> |s3\_origin\_path                            | Return S3 Origin path for given `origin_domain_name`, blank otherwise. see example section below |

Reviewer: ```suggestion
|s3\_origin\_path                            | The S3 origin path if `origin_domain_name` is specified in the resource parameters. |
```

      it { should_not exist }
    end

### s3\_origin\_path

> The `s3_origin_path` either return origin path or blank in case of no origin configured.
> 
>       describe aws_cloudfront_distribution(distribution_id: 'EXISTING_DISTRIBUTION_ID', origin_domain_name: 'EXISTING_CF_ORIGIN_NAME_1') do
>         its ('s3_origin_path') { should include '/next' }
>       end
> 
>       describe aws_cloudfront_distribution(distribution_id: 'EXISTING_DISTRIBUTION_ID', origin_domain_name: 'EXISTING_CF_ORIGIN_NAME_2') do
>         its ('s3_origin_path') { should include '/release' }
>       end
> 
> For default origin path
> 
>     describe aws_cloudfront_distribution(distribution_id: 'EXISTING_DISTRIBUTION_ID', origin_domain_name: 'EXISTING_CF_ORIGIN_NAME_3') do
>         its ('s3_origin_path') { should include '' }
>     end

Reviewer: ```suggestion
Use `s3_origin_path` to return an origin path for the specified origin domain name if the origin path is configured, otherwise it returns an empty string.

      describe aws_cloudfront_distribution(distribution_id: 'DISTRIBUTION_ID', origin_domain_name: 'ORIGIN_DOMAIN_NAME') do
        its ('s3_origin_path') { should include '/next' }
      end

      describe aws_cloudfront_distribution(distribution_id: 'DISTRIBUTION_ID', origin_domain_name: 'ORIGIN_DOMAIN_NAME') do
        its ('s3_origin_path') { should include '/release' }
      end

For the default origin path:

    describe aws_cloudfront_distribution(distribution_id: 'DISTRIBUTION_ID', origin_domain_name: 'ORIGIN_DOMAIN_NAME') do
        its ('s3_origin_path') { should include '' }
    end
```
